app/models/project_step.rb

```diff
@@ -135,15 +152,15 @@ def is_finalized_locked?
 
   # The date to use when calculating days_late.
   def on_time_date
-    original_date || scheduled_date
+    original_date || scheduled_start_date
```

Comment:
This method and all the other `_date` methods (`on_time_date, original_date, scheduled_day, calendar_date`) should all also get the `start_` prefix. Can you please also group them all together in the class.